### PR TITLE
Azure backup DNS zone geo code fix and docs

### LINKED ISSUE #279: 🪲 Bug Report - Azure Backup Private DNS Zones for Private Link Missing Geo Codes
## Describe the bug

Azure Backup Private DNS Zones for Private Link Missing Geo Codes

See the following in the ALZ Terraform Module repo where this was raised and has been resolved already:

- https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues/412
- https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/pull/423

## Expected behaviour

Correct geo code inserted into Private DNS Zone Namespace for Azure Backup

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Azure backup DNS zone geo code fix and docs

## This PR fixes/adds/changes/removes

1. Updates `privateDnsZone` & `hubNetworking` module
2. Fixes #279 

### Breaking Changes

Old Azure Backup Private DNS zone will need to be manually removed by users but new, correct, DNS zone will be created on incremental deployments.

## Testing Evidence

Linting & Unit Tests will suffice

image

image

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep [hunk 1]
```diff
@@ -111,14 +111,13 @@ param parPrivateDnsZones array = [
...
   'privatelink.table.cosmos.azure.com'
-  'privatelink.${parLocation}.batch.azure.com'
+  'privatelink.${toLower(parLocation)}.batch.azure.com'
   'privatelink.postgres.database.azure.com'
   'privatelink.mysql.database.azure.com'
   'privatelink.mariadb.database.azure.com'
   'privatelink.vaultcore.azure.net'
   'privatelink.managedhsm.azure.net'
-  'privatelink.${parLocation}.azmk8s.io'
-  'privatelink.${parLocation}.backup.windowsazure.com'
```
Comment (jfaurskov) [inline infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:121]: Shouldn't this still be in there? It's in the parameters file

**Review by jfaurskov — COMMENTED** [review]
Seems to be missing a dns namespace in hubnetworking module

File: infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep (hunk 1 above)
Comment (jtracey93) [reply infra-as-code/bicep/modules/hubNetworking/hubNetworking.bicep:121]: Backup is removed from here as it gets populated by the var logic in the `privateDnsZone` module. But if someone is using a Parmeter file they still need to specify it

**Review by jfaurskov — APPROVED** [review]
LGTM

**Pull request merged**